ui/page/collection/internal/collectionActions/internal/playButton/index.js

```diff
@@ -0,0 +1,28 @@
+import { connect } from 'react-redux';
+import { selectClaimForUri } from 'redux/selectors/claims';
+import { selectUrlsForCollectionId } from 'redux/selectors/collections';
+import ShuffleButton from './view';
+
+const select = (state, props) => {
+  const { collectionId } = props;
+
+  const collectionUrls = selectUrlsForCollectionId(state, collectionId);
+
+  let firstItem;
+  if (collectionUrls) {
+    // this will help play the first valid claim in a list
+    // in case the first urls have been deleted
+    collectionUrls.map((url) => {
+      const claim = selectClaimForUri(state, url);
+      if (firstItem === undefined && claim) {
+        firstItem = claim.permanent_url;
+      }
+    });
+  }
+
+  return {
+    uri: firstItem,
+  };
+};
+
```

Comment:
- [ ] Performance concerns + wrong function used

Performance should drop significantly here if you have a super-huge, since the state-to-map is a hot area and we are looping the entire array.

Looking at the intent, it shouldn't be using `map` for various wrong reasons.  Just use a for-loop so we can short-circuit when the entry is found.  If many undefined claims are expected at the top, then cache this inside a `createCachedSelector` (although I doubt this is the case).

<sub>https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map</sub>

<img width="536" alt="image" src="https://user-images.githubusercontent.com/64950861/173532491-49b8bf0d-d5e6-4afb-80de-060a4e26e8c1.png">